src/test/util_string_tests.cpp

```diff
@@ -8,21 +8,22 @@
 #include <test/util/setup_common.h>
 
 using namespace util;
+using util::detail::CheckNumFormatSpecifiers;
 
 BOOST_AUTO_TEST_SUITE(util_string_tests)
 
 // Helper to allow compile-time sanity checks while providing the number of
 // args directly. Normally PassFmt<sizeof...(Args)> would be used.
 template <unsigned NumArgs>
-inline void PassFmt(util::ConstevalFormatString<NumArgs> fmt)
+inline void PassFmt(ConstevalFormatString<NumArgs> fmt)
```

Comment:
nit: given that `PassFmt` is local to `util_string_tests`, I think `static` would make more sense here (and the other helpers which sometimes have inline and sometimes nothing):
```suggestion
static void PassFmt(ConstevalFormatString<NumArgs> fmt)
```

Reply:
They are also template functions, so implicitly inline, and arguably local already. Holding off on this for now.

Reply:
I find it inconsistent that `PassFmt` and `FailFmtWithError` are marked inline, but the new `TfmFormatZeroes` is not.
I'm fine with not making them static, but we could at least remove the redundant `inline`s - not critical, I won't block, would just prefer consistency.